include_physics/ideal_MHD.h

```diff
@@ -606,62 +607,70 @@ inline void add_surface_terms(REAL time, uint ix, uint iy, uint iz, const global
 
   REAL alx, arx, aly, ary, alz, arz;
 
+#ifndef USE_PERIODIC_X
+
   if (check_bound_xr(ix, 1)) {
     calc_hll_speeds(um, ub, &alx, &arx, 0);
     calc_flux_f(um, fluxm);
     calc_flux_f(ub, fluxb);
     calc_num_flux(alx, arx, um, ub, fluxm, fluxb, flux);
     for(i = 0; i < NUM_CONSERVED_VARS; i++)
-      du_dt[i] -= (REAL)(M_INV[0] / DX) * (flux[i] - fluxm[i]);
+      du_dt[i] -= (REAL)(M_INV_B[0] / DX) * (flux[i] - fluxm[i]);
```

Comment:
It might be more clear to use `M_INV_X` instead of `M_INV_B` (although they are identical in this case).